### ISSUE TITLE
Cannot read empty arrays in var-length int dataset

### ISSUE THREAD

**Issue by JCzogalla** [jamesmudd/jhdf#247]
The dataset from the h5py docu https://docs.h5py.org/en/stable/special.html#arbitrary-vlen-data cannnot be read by jhdf. Similar for a smaller example with an empty array in the middle:

```
with h5py.File('test.hdf5', 'w') as f:
    dt = h5py.vlen_dtype(np.dtype('int32'))
    dset = f.create_dataset('vlen_int', (3,), dtype=dt)
    dset[0] = [1,2,3]
    dset[1] = []
    dset[2] = [1,2,3,4,5]
```
	
this exception is thrown:

```
Exception in thread "main" io.jhdf.exceptions.HdfException: Error reading global heap at address 0
	at io.jhdf.GlobalHeap.<init>(GlobalHeap.java:76)
	at io.jhdf.dataset.VariableLengthDatasetReader.lambda$readDataset$0(VariableLengthDatasetReader.java:58)
	at java.util.HashMap.computeIfAbsent(HashMap.java:1127)
	at io.jhdf.dataset.VariableLengthDatasetReader.readDataset(VariableLengthDatasetReader.java:57)
	at io.jhdf.object.datatype.VariableLength.fillData(VariableLength.java:101)
	at io.jhdf.dataset.DatasetReader.readDataset(DatasetReader.java:64)
	at io.jhdf.dataset.DatasetBase.getData(DatasetBase.java:128)
Caused by: io.jhdf.exceptions.HdfException: Global heap signature 'GCOL' not matched, at address 0
	at io.jhdf.GlobalHeap.<init>(GlobalHeap.java:47)
	... 7 more
```

The reason is that h5py encodes empty arrays with global heap index 0 and address 0.
HDFView and h5py can read the result.

Changing VariableLengthDatasetReader l. 56 ff to

```
for (GlobalHeapId globalHeapId : getGlobalHeapIds(buffer, type.getSize(), hdfFc, getTotalPoints(dimensions))) {
	if (globalHeapId.getIndex() == 0) {
		elements.add(ByteBuffer.allocate(0)); //TODO: constant?
	} else {
		GlobalHeap heap = heaps.computeIfAbsent(globalHeapId.getHeapAddress(),
				address -> new GlobalHeap(hdfFc, address));

		ByteBuffer bb = heap.getObjectData(globalHeapId.getIndex());
		elements.add(bb);
	}
}
```
fixes the problem.

**Comment by jamesmudd** [comment]
Thanks for finding this issue and the propose solution. I will take a look.

**Comment by jamesmudd** [comment]
Merged PR #248 that resolves this issue.

**Comment by jamesmudd** [comment]
Fixed in the https://github.com/jamesmudd/jhdf/releases/tag/v0.6.0 release
